File: app/src/main/java/com/orafaaraujo/androiddevconference2017/di/ApplicationComponent.java
package com.orafaaraujo.androiddevconference2017.di;

import android.support.annotation.NonNull;

import com.orafaaraujo.androiddevconference2017.DevConferenceApplication;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Main application {@link Component}.
 */
@Singleton
@Component(modules = {ApplicationModule.class})
public interface ApplicationComponent {

    void inject(@NonNull DevConferenceApplication devConferenceApplication);

}

File: app/src/main/java/com/orafaaraujo/androiddevconference2017/di/ApplicationComponent.kt
package com.orafaaraujo.androiddevconference2017.di

import com.orafaaraujo.androiddevconference2017.DevConferenceApplication
import com.orafaaraujo.androiddevconference2017.ui.MainNewWayDaggerActivity
import dagger.Component
import javax.inject.Singleton

/**
 * Main application [Component].
 */
@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {

    fun inject(devConferenceApplication: DevConferenceApplication)

    fun inject(mainNewWayDaggerActivity: MainNewWayDaggerActivity)
}
